### DISCUSSION TITLE
Mapping TimeZoneInfo Class

### DISCUSSION THREAD

**Discussion by dinmukhamed09** [nhibernate/fluent-nhibernate#448] [General] [answered]
Dear Support,

I have problems with mapping.

In class i'm using timezoneinfo property, but when i mapping i have errors.

Example:
```
public class City : ICity
	{
		public virtual string Name { get; set; }
		public virtual TimeZoneInfo TimeZoneInfo { get; set; }
	}
```

```
public class CityMap : ClassMap<City>
	{
		public CityMap()
		{
			Map(x => x.Name).Not.Nullable();

			References(x => x.TimeZoneInfo).Not.Nullable();
		}
	}
```

I'm trying to use mapping of TimeZoneInfo class. but i have to added to all properties virtual

**Comment by dinmukhamed09** [comment]
I'm use workaround, store TimeZoneInfoID to db, and class property TimeZoneInfo returns TimeZoneInfo.FindSystemTimeZoneById(TimeZoneInfoID)

**Comment by hazzik** [comment, marked as answer]
You would need to implement IUserType. Check this SO question: https://stackoverflow.com/q/1565056/259946

**Reply by dinmukhamed09** [reply]
Hi! Thank you for response, I used workaround and for me it is okay.
